ARCs/arc-0028.md

```diff
@@ -0,0 +1,81 @@
+---
+arc: 28
+title: Algorand Event Log Spec
+status: Draft
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/{TBD}
+---
+
+# Algorand Event Log Spec
+
+> This API is a draft.
+> Some elements may change.
+
+## Summary
+
+A methodology for structured logging by Algorand dapps.
+
+## Abstract
+
+Algorand dapps can use the `log` primitive to attach information about an application call. This ARC proposes the concept of Events, which are merely a way in which data contained in these logs may be categorized and structured.
+
+In short: to emit an Event, a dapp calls `log` with ABI formatting of the log data, and a 4-byte prefix to indicate which Event it is.
+
+## Specification
+
+Each kind of Event emitted by a given dapp has a unique 4-byte identifier. This identifier is derived from its name and the structure of its contents, like so:
+
+### Event Signature
+
+An Event Signature is a utf8 string, comprised of: the name of the event, followed by an open paren, followed by the comma-separated names of the data types contained in the event, followed by a close paren. This follows naming conventions similar to ABI signatures, but does not include the return type.
+
+### Deriving the 4-byte prefix from the Event Signature
+
+To derive the 4-byte prefix from the Event Signature, perform the `sha512/256` hash algorithm on the signature, and select the first 4 bytes of the result.
+
+This is the same process that is used by the [ABI Method Selector as specified in ARC 4](./arc-0004.md#method-selector).
+
+### Sample interpretation of Event log data
+
+An exchange dapp might emit a `Swapped` event with two `uint64` values representing quantities of currency swapped. The event signature would be: `Swapped(uint64,uint64)`.
```

Comment:
should we stick with lowercase names by default?

Reply:
I'm not sure what you mean when you say "by default". Do you simply mean that this example should use a lower-case name? Or that something more should be written into the ARC specifically recommending lower-cased names?

Note that HumbleSwap v2 contracts conform to this ARC, and its event names are upper-cased, which inspired this example.

Reply:
fair enough, for some reason i had in my head it was lowercase names only but A-Za-z are allowed 👍